### PR TITLE
Add ability to upload images from telegram API to imgur and have the bot use the imgur link

### PR DESCRIPTION
**Pull request by Chris-Bitler**
This is related to the discussion in https://github.com/RITlug/teleirc/issues/33 - Since telegram doesn't send the correct mime type for the file uploaded (it sends application/octet-stream instead), in order to allow people to view the images uploaded without having to download them to their own device, it makes sense to re-upload them somewhere else.

In order for this to work, people who intend to use the imgur upload functionality need to use imgur's API site to request a clientID and replace it in the .env file. This is why imgur upload is disabled by default.

### CODE REVIEW

**Comment by Chris-Bitler** [comment]
I will fix the tests that this broke

**Comment by jflory7** [comment]
@Chris-Bitler See if you can fix the build errors and we'll get this merged in. Also, thanks for writing a brief snippet of docs to add into the README. This is helpful.

Also, while the environment variables are nice if you're using Docker, we should also have a way for people to use the configuration file, `config.js`, to configure the Imgur support. Can you make sure that this functionality is included into `config.js.example`?

File: .env.example [hunk 1]
```diff
@@ -12,4 +12,6 @@ SHOW_ACTION_MESSAGE=true
 SHOW_JOIN_MESSAGE=true
 SHOW_KICK_MESSAGE=true
 SHOW_LEAVE_MESSAGE=true
+USE_IMGUR_FOR_IMAGES=false
+IMGUR_CLIENT_ID=0000000000
```
Comment (jflory7) [inline .env.example]: Also, while the environment variables are nice if you're using Docker, we should also have a way for people to use the configuration file, `config.js`, to configure the Imgur support. Can you make sure that this functionality is included into `config.js.example`?

**Comment by Chris-Bitler** [comment]
Tests pass now. I didn't add any tests related to exceptions for the imgur section, because it fails gracefully if it doesn't exist and just uses the telegram file url.

**Comment by jflory7** [comment]
I rebased these commits into ecf15a1b1892373f57314eccd4068d706083f7e5 and pushed it to devel. This is now live! Thanks for contributing, @Chris-Bitler! 🎉 💯 🎉

🌮 🌯

**Pull request closed without merging**